### PR TITLE
Feat/support v8

### PR DESCRIPTION
**Pull request by bjelugbo**
PR for types definition issues

### CODE REVIEW

**Comment by rameshkumar91** [comment]
@bjelugbo 
image
Guess there is a commit from master that needs to be pulled from master

**Comment by christian-bromann** [comment]
Can we rebase this branch and resolve conflicts?

**Comment by christian-bromann** [comment]
Thank you! It is a bummer we don't have any e2e tests. Do you think we should add a simple one that checks if a report is generated successfully as part of this?

**Comment by christian-bromann** [comment]
@wswebcreation any comments?

**Comment by fringd** [comment]
i'm noticing the package.json still has:

```
    "peerDependencies": {
        "@wdio/cucumber-framework": "7.x"
    },
```

you should bump that to 8.x

**Comment by dlitmano** [comment]
New version is 5.0.0?
```
{
    "name": "wdio-cucumberjs-json-reporter",
    "version": "4.4.3",
```

**Comment by dlitmano** [comment]
webdriverio latest dependency is 8.8.2

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: wdio-cucumberjs-json-reporter-e2e
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
```
Comment (christian-bromann) [inline .github/workflows/e2e.yml:10]: Let's have test run on all pull request branches
```suggestion
```

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+defaults:
+  run:
+    working-directory: ./
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [16.x]
```
Comment (christian-bromann) [inline .github/workflows/e2e.yml:21]:
```suggestion
        node-version: [18.x]
```

File: lib/tests/wdio.conf.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import WdioCucumberJsJsonReporter from '../../dist/lib/reporter.js';
+import path from 'path';
+import { fileURLToPath } from 'url';
+const __filename = fileURLToPath( import.meta.url );
+const __dirname = path.dirname( __filename );
+console.log(WdioCucumberJsJsonReporter)
+export const config = {
```
Comment (christian-bromann) [inline lib/tests/wdio.conf.ts:7]:
```suggestion
import path from 'node:path'
import { fileURLToPath } from 'node:url'
import type { Options } from '@wdio/types'

import WdioCucumberJsJsonReporter from '../../lib/reporter.js'

const __filename = fileURLToPath(import.meta.url)
const __dirname = path.dirname(__filename)

export const config: Options.Testrunner = {
```

File: lib/tests/wdio.conf.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    specs: [__dirname +'/e2e/*.feature'],
```
Comment (christian-bromann) [inline lib/tests/wdio.conf.ts:8]:
```suggestion
    specs: [path.join(__dirname, 'e2e', '*.feature')],
```

**Comment by wswebcreation** [comment]
> @wswebcreation any comments?

Sorry,

I normally had a local sample project that I linked. It used the following


## sample.feature
```
@sample
Feature: Sample

  Background:
    Given I have a background

  Scenario: cukes
    Given I have 42 cukes in my belly

  @data-table
  Scenario: transposed table
    When the following table is transposed
      | a | b |
      | 1 | 2 |

  @outline @skipped
  Scenario Outline: eating cucumbers
    Given there are <start> cucumbers
    When I eat <eat> cucumbers
    Then I should have <left> cucumbers

    Examples: These are passing
      | start | eat | left |
      | 12    | 5   | 7    |
      | 20    | 5   | 15   |
      | 25    | 5   | 20   |
```

## sample.copy.feature
```
@sample-copy
Feature: Sample Copy

  Background:
    Given I have a background

  Scenario: cukes
    Given I have 42 cukes in my belly

  @data-table-copy
  Scenario: transposed table
    When the following table is transposed
      | a | b |
      | 1 | 2 |

  @outline-copy @skipped-copy
  Scenario Outline: eating cucumbers
    Given there are <start> cucumbers
    When I eat <eat> cucumbers
    Then I should have <left> cucumbers

    Examples: These are passing
      | start | eat | left |
      | 12    | 5   | 7    |
      | 20    | 5   | 15   |
      | 25    | 5   | 20   |
```

## sample.steps.js
```js
const {Before, Given, When, Then} = require('@cucumber/cucumber');
const cucumberJson = require('wdio-cucumberjs-json-reporter').default;


Given('I have {int} cukes in my belly', async (cukeCount) => {
  await browser.url('https://webdriver.io/')
  await cucumberJson.attach(await browser.takeScreenshot(), 'image/png');
  return true
})
Given(/^I have a background$/, () => {
  return true;
});
When(/^the following table is transposed$/, async (table) => {
  return true;
});

Given('there are {int} cucumbers', async (initialCount) => {
  return true;
})

When('I eat {int} cucumbers', async (eatCount) => {
  return true;
})

Then('I should have {int} cucumbers', async (expectedCount) => {
  return true;
})

Before(async ()=>{
  return true
})
```

**Comment by christian-bromann** [comment]
@bjelugbo thanks, the changes look good to me. I will merge them and will do some testing before the release

**Comment by christian-bromann** [comment]
I hope by next week I get the chance to look into this.

**Pull request merged**